drivers/include/net/netdev2/lorawan.h

```diff
@@ -0,0 +1,225 @@
+}lorawan_sesion_t;
+
+typedef struct sMulticastParams
+{
+    uint32_t Address;
+    uint8_t NwkSKey[16];
+    uint8_t AppSKey[16];
+    uint32_t DownLinkCounter;
+    struct sMulticastParams *Next;
+}MulticastParams_t;
+
+typedef enum eDeviceClass
+{
+    CLASS_A,
+    CLASS_B,
+    CLASS_C,
+}DeviceClass_t;
+
+typedef struct sRx2ChannelParams
+{
+    uint32_t Frequency;
+    uint8_t  Datarate;
+}Rx2ChannelParams_t;
+
+typedef struct sLoRaMacParams
+{
+    int8_t ChannelsTxPower;
+    int8_t ChannelsDatarate;
+    uint32_t MaxRxWindow;
+    uint32_t ReceiveDelay1;
+    uint32_t ReceiveDelay2;
+    uint32_t JoinAcceptDelay1;
+    uint32_t JoinAcceptDelay2;
+    uint8_t ChannelsNbRep;
+    uint8_t Rx1DrOffset;
+    Rx2ChannelParams_t Rx2Channel;
+    uint16_t ChannelsMask[6];
+}LoRaMacParams_t;
+
+
+typedef enum eMcps
+{
+    MCPS_UNCONFIRMED,
+    MCPS_CONFIRMED,
+    MCPS_MULTICAST,
+    MCPS_PROPRIETARY,
+}Mcps_t;
+
+typedef struct {
+    netdev2_t netdev;                       /**< @ref netdev2_t base class */
+    lorawan_sesion_t lorawan;
+    uint8_t *dev_eui;
+    uint8_t *app_key;
+    uint8_t *app_eui;
+    uint16_t dev_nonce;
+    MulticastParams_t *MulticastChannels;
+    DeviceClass_t LoRaMacDeviceClass;
+    bool RepeaterSupport;
+    uint8_t LoRaMacPayload[LORAMAC_PHY_MAXPAYLOAD];
+    uint32_t uplink_counter;
+    uint32_t DownLinkCounter;
+    bool IsUpLinkCounterFixed;
+    bool IsRxWindowsEnabled;
+    uint32_t AdrAckCounter;
+    bool NodeAckRequested;
+    bool SrvAckRequested;
+    bool MacCommandsInNextTx;
+    uint8_t MacCommandsBufferIndex;
+    uint8_t MacCommandsBufferToRepeatIndex;
+    uint8_t MacCommandsBuffer[LORA_MAC_COMMAND_MAX_LENGTH];
+    uint8_t MacCommandsBufferToRepeat[LORA_MAC_COMMAND_MAX_LENGTH];
+    uint16_t ChannelsMaskRemaining[6];
+    LoRaMacParams_t LoRaMacParams;
+    LoRaMacParams_t LoRaMacParamsDefaults;
+    uint8_t ChannelsNbRepCounter;
+    uint8_t MaxDCycle;
+    uint16_t AggregatedDCycle;
+    TimerTime_t AggregatedLastTxDoneTime;
+    TimerTime_t AggregatedTimeOff;
+    bool DutyCycleOn;
+    uint8_t Channel;
+    uint8_t LastTxChannel;
+    TimerEvent_t TxDelayedTimer;
+    TimerEvent_t RxWindowTimer1;
+    TimerEvent_t RxWindowTimer2;
+    uint32_t RxWindow1Delay;
+    uint32_t RxWindow2Delay;
+    TimerEvent_t AckTimeoutTimer;
+    uint8_t ack_timeout_retries; /**< Current number of ACK attempts */
+    uint8_t ack_timeout_retries_counter; /**< Number of ACK retries */
+    TimerTime_t TxTimeOnAir;
+    uint16_t join_request_trials; /**< Attempts of join requests */
+    uint8_t rx_slot; /**< Current RX slot */
+
+    uint8_t demod_margin;
+    uint8_t number_of_gateways;
+
+    bool join_req_sent; /**< A join request packet was sent */
+    //TODO: Port to be removed...
+    uint8_t port; 
+    uint8_t *Buffer;
+    uint8_t BufferSize;
+    bool RxData;
+    //Frame variables
+    bool b_tx;
+    bool b_rx;
+    bool link_check;
+    bool join_req;
+    gnrc_pktsnip_t *pkt;
```

Comment:
The GNRC LoRaWAN code lives in the sys/net/gnrc folder, but all variables are handled in this netdev2_lorawan_t.
Indeed it's GNRC dependent.

In this case, where should these variables live? (E.g a gnrc_netdev2_lorawan_t object?)